### ISSUE TITLE
Padding for buttons

### ISSUE THREAD

**Issue by i-am-rita** [OpenRefine/OpenRefine#5706]
1. A little padding on the buttons on the "New Computer, Web addresses, Clipboard and Google data pages" like the "New connection" button in the "Database" page would enhance the UI and make the CTA aesthetically pleasing.
<img width="142" alt="Screen Shot 2023-03-20 at 1 41 32 PM" src="https://user-images.githubusercontent.com/69390818/226351744-9481f778-9515-444e-987d-80a62ed1bfae.png">
<img width="279" alt="Screen Shot 2023-03-20 at 1 41 46 PM" src="https://user-images.githubusercontent.com/69390818/226351835-4707dadf-1829-41d0-81f0-03f91f87d7ab.png">

The buttons should/can be padded like this:
<img width="215" alt="Screen Shot 2023-03-20 at 1 42 40 PM" src="https://user-images.githubusercontent.com/69390818/226351971-f788e398-733c-48e0-884d-e321013dd66a.png">

2. In Google Data page, the input field and next button need some margin because the button is leaning on the input field, 
<img width="665" alt="Screen Shot 2023-03-20 at 2 01 09 PM" src="https://user-images.githubusercontent.com/69390818/226353088-98dcbcd7-682c-46bd-8f48-16dc3a7f2a12.png">
<img width="137" alt="Screen Shot 2023-03-20 at 2 25 58 PM" src="https://user-images.githubusercontent.com/69390818/226353750-5496f3b4-c469-402c-9348-5c70e24ae52d.png">

Also, I think : 
- The "Import a public Google Spreadsheet by its URL:" can be a placeholder in the input field.
- There should be padding in the input field and enough width to contain a long url.

**Comment by ayushrai206** [comment]
i would like to work on this,please assign this to me

**Comment by wetneb** [comment]
@ayushrai206 I would first wait for others to discuss the proposed changes and check if they are consensual.

**Comment by i-am-rita** [comment]
okay. @wetneb  I hope to be assigned to the issue because I want it to be my first contribution to OpenRefine project.

**Comment by Abbe98** [comment]
Note that the "New connection" is the odd one and that on the main branch it looks just like the other buttons.

**Comment by i-am-rita** [comment]
> > A little padding on the buttons on the "New Computer, Web addresses, Clipboard and Google data pages" like the "New connection" button in the "Database" page would enhance the UI and make the CTA aesthetically pleasing.
> 
> @i-am-rita I'm struggling a little to understand what you are proposing here - could you show what the difference before and after the change is?
>> Hi @ostephens the image  below is what the buttons currently look like and if padding is added, they'll look like the "New Connection" button
<img width="279" alt="Screen Shot 2023-03-20 at 1 41 46 PM" src="https://user-images.githubusercontent.com/69390818/226686102-cf31455c-420a-44f2-b6b1-908b0b5b43a4.png">
<img width="215" alt="Screen Shot 2023-03-20 at 1 42 40 PM" src="https://user-images.githubusercontent.com/69390818/226686168-10e87a87-1953-4576-8bf9-0ee5a2d9eb34.png">

> > In Google Data page, the input field and next button need some margin because the button is leaning on the input field,
> 
> I think this is a clear proposal and I completely support it - the current layout is not good
>>Thanks.
> > The "Import a public Google Spreadsheet by its URL:" can be a placeholder in the input field.
> 
> I definitely think we could improve the UX here, but I don't agree with this proposal. This text should is a label or instruction to the user - not placeholder text. If we were to put the placeholder text in the field it can only be to enhance information already visible to the user (as the placeholder text is not visible as soon as the user starts entering information). See also https://www.nngroup.com/articles/form-design-placeholders/
I agree with this, and looking at it again, I just realized that it's an instruction. Thank you
> > There should be padding in the input field and enough width to contain a long url.
> 
> Agreed but can you look at what the typical length of a Google docs public URL might be so we get this length more specific than just "long url"?
Some google docs url can be long, I'm proposing we make the input field a bit wide and probably add "..." to show that there is more because if you write in the input field, the beginning of the link doesn't show. Below is a screenshot: 
<img width="522" alt="Screen Shot 2023-03-21 at 6 11 41 PM" src="https://user-images.githubusercontent.com/69390818/226689179-a4bd6700-cbd5-4b0b-a999-04e99555f81a.png">

> > Lastly, an increase in the font-size of the texts/urls to be entered in the input fields.
> 
> No specific objection to this but I would appreciate some information on why this is needed and what size would be appropriate? At the moment the font size used for input seems to be in line with the font size being used for display. Is there a reason the input font size would be bigger than the font size used in the rest of the UI?
>> I just saw that the font of the rest of the Ui is similar to the input field. So, i think we can leave the font-size the way it is until there's a proposal for a new UI.

**Comment by tfmorris** [comment]
> There should be padding in the input field and enough width to contain a long url.

Note that these URLs are almost always going to be pasted from the cut buffer rather than typed by the user. I'm not opposed to making the Google Data url field bigger, but I don't think it needs to accommodate a full length URL.

@i-am-rita Can you please adjust the quoting on your last [comment](https://github.com/OpenRefine/OpenRefine/issues/5706#issuecomment-1478319873)? It looks like you may have your replies indented instead of dedented like the normal quoting convention, which makes it difficult to read.

**Comment by i-am-rita** [comment]
> > There should be padding in the input field and enough width to contain a long url.
> 
> Note that these URLs are almost always going to be pasted from the cut buffer rather than typed by the user. I'm not opposed to making the Google Data url field bigger, but I don't think it needs to accommodate a full length URL.
> 

Okay
> @i-am-rita Can you please adjust the quoting on your last [comment](https://github.com/OpenRefine/OpenRefine/issues/5706#issuecomment-1478319873)? It looks like you may have your replies indented instead of dedented like the normal quoting convention, which makes it difficult to read.

Thank you @tfmorris 😊 I'm new to contributing to open source projects and learning new stuff everyday. Thanks again.

**Comment by ostephens** [comment]
>if padding is added, they'll look like the "New Connection" button

The New Connection button is different in styling and in a different context. I still can't tell what adding padding will do (assuming that this doesn't change the color of the button!)

**Comment by i-am-rita** [comment]
> > if padding is added, they'll look like the "New Connection" button
> 
> As mentioned by @Abbe98 the "New Connection" button is the one styled in a non-standard way currently. If the proposal is to have the buttons styled to look like the "New Connection" button I'm against this in isolation - we should aim for consistent styling across the application

I agree with aiming for consistent styling. My proposal on adding padding is so that the buttons will look aesthetically pleasing and to enhance user experience.

Hi @wetneb I understand that to make changes to OpenRefine, one would have to consider a lot including documentations and the videos explaining how to use OpenRefine but I would like to ask if designing a new interface with the documentation and videos in mind is something the OpenRefine team hope to do someday.

**Comment by ostephens** [comment]
>I agree with aiming for consistent styling. My proposal on adding padding is so that the buttons will look aesthetically pleasing and to enhance user experience.

In which case I'm still not understanding what add the padding will actually do - you said:
>they'll look like the "New Connection" button

Can you be clearer about what is actually going to change?

**Comment by i-am-rita** [comment]
> > I agree with aiming for consistent styling. My proposal on adding padding is so that the buttons will look aesthetically pleasing and to enhance user experience.
> 
> In which case I'm still not understanding what add the padding will actually do - you said:
> 
> > they'll look like the "New Connection" button
> 
> Can you be clearer about what is actually going to change?

Just the padding.

**Comment by ostephens** [comment]
I guess I still don't understand what the change will look like the user which is what I was trying to get to. But probably much easier to review from a PR to be honest.

**Comment by ostephens** [comment]
@i-am-rita apologies - I'm not sure if your discussion here is just to agree the design or you are also going to do the changes? If you are just proposing a design for others to implement then I think these proposals need to be more explicit. If you are planning on doing the work yourself I'd suggest it would be easier to comment on a PR at this point as that would have the explicit information

**Comment by i-am-rita** [comment]
> @i-am-rita apologies - I'm not sure if your discussion here is just to agree the design or you are also going to do the changes? If you are just proposing a design for others to implement then I think these proposals need to be more explicit. If you are planning on doing the work yourself I'd suggest it would be easier to comment on a PR at this point as that would have the explicit information

I am actually planning on doing the work myself that's why I asked above if there're plans for a redesign of OpenRefine's UI.
If there is going to be a redesign, I would happily work with the team.

**Comment by ostephens** [comment]
I'm not aware of any immediate large-scale redesign of the UI (although people have mentioned this as an aim for v4 of OR) and I would certainly go ahead and make a PR for these changes as soon as you have the time

**Comment by elebitzero** [comment]
Hey, @i-am-rita

I would suggest creating one issue per item (one issue for the button padding and one issue for the Google URL input UX).


> Issue #1 padding on buttons on Create Project page

I attempted to increase the UI button size. At the time, there was some hesitation about changing the button size across the entire webapp, so I shelved it.  Ref: https://github.com/OpenRefine/OpenRefine/pull/4957

My starting point was to update the OR button style to match the current jQuery UI's default button size. (See https://jqueryui.com/button/)


> Issue #2 Google URL input

In Create Project->Web Addresses (URLs), the URL input textbox is using 80% of the panel width, so it takes advantage of the available space. I would do the same for the Google URL.

**Comment by SourabhMalviya9090** [comment]
I’m interested in working on this and would love to contribute. Could you please assign it to me?
